### ISSUE TITLE
Upgrade from 1.25.7 to 1.26.2 fails with nginx ingress upgrade

### ISSUE THREAD

**Issue by xhejtman** [rancher/rke2#4008]
**Environmental Info:**
RKE2 Version: 1.26.2-rke2r1

After upgrade from 1.25.7 to 1.26.2 via rke2, helm upgrade of rke2-ingress-nginx fails with:
```
helm_v3 upgrade --set-string 'global.clusterCIDR=10.42.0.0/16\,2001:718:xxx:xxx:8:2::/96' --set-string global.clusterCIDRv4=10.42.0.0/16 --set-string global.clusterCIDRv6=2001:718:xxx:xxx:8:2::/96 --set-string global.clusterDNS=10.43.0.10 --set-string global.clusterDomain=cluster.local --set-string global.rke2DataDir=/var/lib/rancher/rke2 --set-string 'global.serviceCIDR=10.43.0.0/16\,2001:718:xxx:xxx:8:3::/112' rke2-ingress-nginx /tmp/rke2-ingress-nginx.tgz --values /config/values-10_HelmChartConfig.yaml
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: resource mapping not found for name: "rke2-ingress-nginx-controller" namespace: "kube-system" from "": no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta2"
```

clearly autocaling/v2 is the only version available in 1.26.

**Comment by xhejtman** [comment]
I wasn't as it stuck in upgrade process, but it was testing cluster only, so I deleted the cluster and run again from scratch and if disabling autoscaler for both controller and default backend, I can run it. For upgrade process, it seems to be required to first disable autoscaler.

I know that helm has problems, if deployment somehow fails, to deploy updated version again. 

I saw that even nginx chart 4.3.0 is not updated to 1.26 version though.

**Comment by brandond** [comment]
I believe this was resolved by recent chart updates; moving to to-test to confirm.

**Comment by ShylajaDevadiga** [comment]
@xhejtman Would you mind sharing the config you have used before installing? I am not able to reproduce this issue. 
1. With default config, upgrade is successful
2. Set autoscaling to true as below does not seem to fail as it is a Daemonset by default 
```
apiVersion: helm.cattle.io/v1
kind: HelmChartConfig
metadata:
  name: rke2-ingress-nginx
  namespace: kube-system
spec:
  valuesContent: |-
    controller:
      autoscaling:
        enabled: true
```
OR as Deployment
```
apiVersion: helm.cattle.io/v1
kind: HelmChartConfig
metadata:
  name: rke2-ingress-nginx
  namespace: kube-system
spec:
  valuesContent: |-
    controller:
      kind: Deployment
      autoscaling:
        enabled: true
```
From the pod logs for both the above configs
```
+ helm_v3 upgrade --set-string global.clusterCIDR=10.42.0.0/16 --set-string global.clusterCIDRv4=10.42.0.0/16 --set-string global.clusterDNS=10.43.0.10 --set-string global.clusterDomain=cluster.local --set-string global.rke2DataDir=/var/lib/rancher/rke2 --set-string global.serviceCIDR=10.43.0.0/16 rke2-ingress-nginx /tmp/rke2-ingress-nginx.tgz
Upgrading rke2-ingress-nginx
Release "rke2-ingress-nginx" has been upgraded. Happy Helming!
```

**Comment by xhejtman** [comment]
I had:
```
autoscaling:
        enabled: true
        minReplicas: 1
        maxReplicas: 2
```

the problem was that autoscaler changed api in 1.26 version, but as this is already resolved in nginx charts, it is not reproducible any more I guess.

**Comment by ShylajaDevadiga** [comment]
Thank you @xhejtman. Closing issue as resolved.
Upgrade from latest released version to rke2 version v1.26.8-rc1+rke2r1 looks good.
